sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -39,20 +39,18 @@ import org.apache.spark.sql.util.CaseInsensitiveStringMap
 /**
  * A [[TableCatalog]] that translates calls to the v1 SessionCatalog.
  */
-class V2SessionCatalog(sessionState: SessionState) extends TableCatalog {
+class V2SessionCatalog(catalog: SessionCatalog, conf: SQLConf) extends TableCatalog {
+
+  def this(sessionState: SessionState) = this(sessionState.catalog, sessionState.conf)
```

Comment:
Is this still used? Should we convert callers to use the new constructor instead?

Reply:
It's mostly for
```
def this() = this(SparkSession.active.sessionState)
```
Due to the limitation of Scala class constructor, I need to call `SparkSession.active.sessionState` twice if we remove the constructor here.

Reply:
Where is the no-arg constructor called? I don't see it in master.

Reply:
I don't think the no-arg constructor is needed after this commit. When this class was created by the catalog loader, it needed to have a no-arg constructor. After this commit, this class is only instantiated internally by Spark. We can remove all of these constructors.